stackslib/src/burnchains/db.rs

```diff
@@ -323,9 +323,17 @@ impl<'a> BurnchainDBTransaction<'a> {
             &u64_to_sql(header.num_txs)?,
             &u64_to_sql(header.timestamp)?,
         ];
-        match self.sql_tx.execute(sql, args) {
-            Ok(_) => Ok(self.sql_tx.last_insert_rowid()),
-            Err(e) => Err(e.into()),
+        let affected_rows = self.sql_tx.execute(sql, args)?;
+        if affected_rows == 0 {
+            // This means a duplicate entry was found and the insert operation was ignored
+            debug!(
+                "Duplicate entry for block_hash: {}, insert operation ignored.",
+                header.block_hash
+            );
+            Ok(-1)
+        } else {
+            // A new row was inserted successfully
+            Ok(self.sql_tx.last_insert_rowid())
```

Comment:
I think you can actually remove the returned rowid, since I don't think it's used anywhere.  This can just return `Ok(())`.

Reply:
Agreed, I thought about changing the return type too. Will make this change.

Reply:
Do you think the log about the duplicate entry is useful to keep, or should we just simplify it to:
```rust
self.sql_tx.execute(sql, args).map(|_| Ok(()))?
```

Reply:
I think keeping the log is a good idea.